code/go/pkg/validator/validator_test.go

```diff
@@ -193,6 +193,13 @@ func TestValidateFile(t *testing.T) {
 				"field vars.0: Additional property secret is not allowed",
 			},
 		},
+		"bad_secret_vars_v3": {
+			"manifest.yml",
+			[]string{
+				"field vars.0: variable identified as possible secret, secret parameter required to be set to true or false",
+				"field vars.1: variable identified as possible secret, secret parameter required to be set to true or false",
```

Comment:
> This is the kind of errors package developers will see when upgrading to Package Spec 3.0.2.

Isn't this an enhancement? I'm surprised a new feature would fall into a patch version.

I'm also hesitant to jump right into _requiring_ `secret` when it's still new functionality  (sure, a dev can set `secret: false` but the way the error's presented, we're encouraging to set `secret: true`). Based on a recent issue I ran into trying to set `secret: true` (see https://github.com/elastic/kibana/issues/154715#issuecomment-1792869195), I think it's worth adopting on a smaller set of integrations first.

Should using `secret` be encouraged, especially with new integrations? Definitely! But _required_? I'm not so sure.

Reply:
Hey @ebeahan,

> I'm surprised a new feature would fall into a patch version.

Starting on Package Spec 3 we are not fully following semver. I understand this can be surprising. We use minors now for compatibility with the stack, and this change doesn't affect the compatibility with the stack, it only enhances validations. Secrets were already supported in the spec since 2.9.0.

It is an enhancement in the sense that it can prevent security risks. We have also considered as enhancements other "breaking" validations that we have introduced in the past to avoid problematic situations. We can reconsider this in any case if needed.

> a dev can set `secret: false` but the way the error's presented, we're encouraging to set `secret: true`

I am open to suggestions regarding the error message.

> Based on a recent issue I ran into trying to set secret: true (see https://github.com/elastic/kibana/issues/154715#issuecomment-1792869195)

This is an issue found in an snapshot build, that was fixed. We can still delay the introduction of this change though, 3.0.2 hasn't been released yet.

Reply:
> Starting on Package Spec 3 we are not fully following semver. I understand this can be surprising. We use minors now for compatibility with the stack, and this change doesn't affect the compatibility with the stack, it only enhances validations. Secrets were already supported in the spec since 2.9.0.

Updating versioning guidelines in the README taking recent changes into account https://github.com/elastic/package-spec/pull/667